### DISCUSSION TITLE
Naming conventions for Stargate v2

### DISCUSSION THREAD

**Discussion by jeffreyscarpenter** [stargate/stargate#1532] [RFCs]
We've been using the following conventions during the course of Stargate v2 development work in order to avoid confusion with v1. Any of these conventions might be revisited at the point in time where v2 becomes the main version of the project.

- GitHub Issues/PRs/Discussions are tagged with the `stargate-v2` label
- The main Git branch is `v2.0.0`. Most work is done in feature branches off of `v2.0.0`, typically named according to the pattern `v2/<dev>/<feature>`, for example `v2/jeff/docker-images`.
- New maven modules are named according to the pattern `sgv2-...`, for example `sgv2-restapi`. You'd only be able to see these on the `v2.0.0`. Existing Maven modules such as `restapi` are generally preserved during development so we can readily test against them to validate v2 implementations have the correct behavior.
  - (Note added 2/7/2022) For the `v2.0.0` GA release, the `sgv2-` prefix will be removed from newer module names, and obsolete modules from Stargate v1 will be removed entirely.
- Docker images will be published under the `stargateio` org, coordinator node images will be named accordingly, for example: `coordinator-4_0` (equivalent to current `stargate-4_0`), `coordinator-3_11` (vs current `stargate-3_11`), `coordinator-dse-68` (vs current `stargate-dse-68`), `sgv2-restapi` (new), etc. Images will be tagged with build number as they are now
- Releases of Stargate v2 will be tagged according to the pattern `v2.0.0-ALPHA-1`, `v2.0.0-ALPHA-2`, `v2.0.0-BETA-1`, until the GA release of v2 which will be `v2.0.0`.

**Comment by dougwettlaufer** [comment]
For `restapi` (which is net new so we probably don't need the prefix) do we want that in its own docker repo or do we want to group them together? Could do something like `stargateio/services-restapi`, `stargateio/services-graphqlapi`, ...

**Reply by jeffreyscarpenter** [reply]
Removing `sgv2-` prefix from docker images
